src/System.Net.Http/src/System/Net/Http/Unix/CurlCallbacks.cs

```diff
@@ -354,49 +362,52 @@ private static bool TryParseStatusCode(string responseHeader, int statusCodeStar
             long timeoutInMilliseconds,
             IntPtr context)
         {
-            // TODO: Handle <=0 values for the timeout
-            Task.Delay((int)timeoutInMilliseconds).ContinueWithStandard(context, (t, c) => CurlTimerElapsed((IntPtr)c));
-
-            return CURLMcode.CURLM_OK;
-        }
-
-        private static void CurlTimerElapsed(
-            IntPtr context)
-        {
+            SafeCurlMultiHandle multiHandle;
             try
             {
                 GCHandle gch = GCHandle.FromIntPtr(context);
-                SafeCurlMultiHandle multiHandle = (SafeCurlMultiHandle)gch.Target;
-                CheckForCompletedTransfers(multiHandle, -1, 0, IntPtr.Zero);
+                multiHandle = (SafeCurlMultiHandle)gch.Target;
             }
             catch
             {
                 // CurlHandler was probably disposed
+                return CURLMcode.CURLM_OK;
             }
-        }
 
-        private static void CheckForCompletedTransfers(SafeCurlMultiHandle multiHandle, int socketFd, int eventBitMask, IntPtr statePtr)
-        {
-            int runningTransfers;
-            // TODO: Revisit the lock and see if serialization is really needed
             lock (multiHandle)
             {
-                int result = Interop.libcurl.curl_multi_socket_action(multiHandle, socketFd, eventBitMask,
-                    out runningTransfers);
-                if (result != CURLMcode.CURLM_OK)
-                {
-                    throw new HttpRequestException(SR.net_http_client_execution_error, GetCurlException(result, true));
-                }
+                int dueTime = (-1 == timeoutInMilliseconds) ? Timeout.Infinite : (int)timeoutInMilliseconds;
```

Comment:
Timeout.Infinite is -1, so this line shouldn't be necessary.  The docs explicitly state this value, e.g. https://msdn.microsoft.com/en-us/library/vstudio/system.threading.timeout.infinite(v=vs.100).aspx
```
The value of this constant is -1.
```
So this can all be collapsed to just:
```C#
lock (multiHandle)
{
     multiHandle.Timer.Change((int)timeoutInMilliseconds, Timeout.Infinite);
}
```